src/main/java/cryptator/CryptaOperator.java

```diff
@@ -8,81 +8,84 @@
  */
 package cryptator;
 
-import java.math.BigInteger;
-import java.util.function.BinaryOperator;
-
 import org.chocosolver.solver.expression.discrete.arithmetic.ArExpression;
+import org.chocosolver.solver.expression.discrete.relational.ReExpression;
 import org.chocosolver.solver.variables.IntVar;
 import org.chocosolver.util.tools.VariableUtils;
 
+import java.math.BigInteger;
+import java.util.function.BinaryOperator;
+
 /**
  * @see https://en.wikipedia.org/wiki/Relational_operator
  */
 public enum CryptaOperator {
-	ADD("+", (a, b) -> a.add(b), (a, b) -> a.add(b)),
-	SUB("-", (a, b) -> a.subtract(b), (a, b) -> a.sub(b)), 
-	MUL("*", (a, b) -> a.multiply(b), (a, b) -> a.mul(b)), 
-	DIV("//", (a, b) -> a.divide(b), (a, b) -> a.div(b)),
-	FDIV("/", (a, b) -> fdiv(a, b), (a, b) -> fdiv(a, b)),
-	MOD("%", (a, b) -> a.mod(b), (a, b) -> a.mod(b)), 
-	POW("^", (a, b) -> a.pow(b.intValue()), (a, b) -> a.pow(b)), 
-	ID("", (a, b) -> BigInteger.ZERO, (a, b) -> null),
-	EQ("=", (a, b) -> toBigInt(a.compareTo(b) == 0), (a, b) -> a.eq(b)), 
-	NE("!=", (a, b) -> toBigInt(a.compareTo(b) != 0), (a, b) -> a.ne(b)), 
-	LT("<", (a, b) -> toBigInt(a.compareTo(b) < 0), (a, b) -> a.lt(b)), 
-	GT(">", (a, b) -> toBigInt(a.compareTo(b) > 0), (a, b) -> a.gt(b)), 
-	LE("<=", (a, b) -> toBigInt(a.compareTo(b) <= 0), (a, b) -> a.le(b)), 
-	GE(">=", (a, b) -> toBigInt(a.compareTo(b) >= 0), (a, b) -> a.ge(b));
+    ADD("+", (a, b) -> a.add(b), (a, b) -> a.add(b)),
+    SUB("-", (a, b) -> a.subtract(b), (a, b) -> a.sub(b)),
+    MUL("*", (a, b) -> a.multiply(b), (a, b) -> a.mul(b)),
+    DIV("//", (a, b) -> a.divide(b), (a, b) -> a.div(b)),
+    FDIV("/", (a, b) -> fdiv(a, b), (a, b) -> fdiv(a, b)),
+    MOD("%", (a, b) -> a.mod(b), (a, b) -> a.mod(b)),
+    POW("^", (a, b) -> a.pow(b.intValue()), (a, b) -> a.pow(b)),
+    ID("", (a, b) -> BigInteger.ZERO, (a, b) -> null),
+    EQ("=", (a, b) -> toBigInt(a.compareTo(b) == 0), (a, b) -> a.eq(b)),
+    NE("!=", (a, b) -> toBigInt(a.compareTo(b) != 0), (a, b) -> a.ne(b)),
+    LT("<", (a, b) -> toBigInt(a.compareTo(b) < 0), (a, b) -> a.lt(b)),
+    GT(">", (a, b) -> toBigInt(a.compareTo(b) > 0), (a, b) -> a.gt(b)),
+    LE("<=", (a, b) -> toBigInt(a.compareTo(b) <= 0), (a, b) -> a.le(b)),
+    GE(">=", (a, b) -> toBigInt(a.compareTo(b) >= 0), (a, b) -> a.ge(b)),
+
+    AND(";", (a, b) -> toBigInt(!a.min(b).equals(BigInteger.ZERO)), (a, b) -> ((ReExpression) a).and((ReExpression) b));
```

Comment:
The evaluation is wrong, for instance with a<0 and b=0.

Reply:
Not yet, but it could happen because the function takes integer as argument. 
It is simpler to directly write a function that is correct for any argument than to wait for a bug to appear. 

The boolean semantic (in C/Java) is :
- false is 0.
- Any other value is true.

Reply:
Ok, I see, I correct it